src/Extensions/Logging/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.Omex.Extensions.Logging.Replayable;
+
+namespace Microsoft.Omex.Extensions.Logging
+{
+	/// <summary>
+	/// Extension methods for the <see cref="IServiceCollection"/> class
+	/// </summary>
+	public static class ServiceCollectionExtensions
+	{
+		/// <summary>
+		/// Add IServiceContext to ServiceCollection
+		/// </summary>
+		public static IServiceCollection AddOmexServiceContext<TServiceContext>(this IServiceCollection serviceCollection)
+			where TServiceContext : class, IServiceContext
+		{
+			serviceCollection.TryAddTransient<IServiceContext, TServiceContext>();
+			return serviceCollection;
+		}
+
+
+		/// <summary>
+		/// Adds Omex event logger to the factory
+		/// </summary>
+		/// <param name="builder">The extension method argument</param>
+		public static ILoggingBuilder AddOmexLogging(this ILoggingBuilder builder)
+		{
+			builder.Services.AddOmexLogging();
+			return builder;
+		}
+
+
+		/// <summary>
+		/// Adds Omex event logger to the factory
+		/// </summary>
+		/// <param name="serviceCollection">The extension method argument</param>
+		/// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained</returns>
+		public static IServiceCollection AddOmexLogging(this IServiceCollection serviceCollection)
+		{
+			serviceCollection
```

Comment:
Nitpick: can be formatted on a single line, which might be clearer given that the others are formatted on a single line.

Reply:
I will resolve it in the follow-up pr https://github.com/microsoft/Omex/pull/168